src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/MapCode/MapCodeEndpoint.cs

```diff
@@ -174,7 +174,8 @@ private async Task<bool> TryMapCodeAsync(
                 // as the code to map. The client is currently implemented using this behavior, but if it
                 // ever changes, we'll need to update this code to account for it (i.e., take into account
                 // focus location URIs).
-                Debug.Assert(location.Uri == documentContext.Uri);
+                // TODO(toddgrun): switch back to == when roslyn implementation of DocumentUri.operator== is available on ci
```

Comment:
>// TODO(toddgrun): switch back to == when roslyn implementation of DocumentUri.operator== is available on ci [](http://example.com/codeflow?start=16&length=108)

There were four places I found where I put this comment and switched the code from using operator== to calling Equals. I just added the operator== implementation to the DocumentUri class in roslyn yesterday, so it can't be used until:

1) Razor updates to a roslyn package that includes that
2) (I think) The ci has a roslyn installed that has that.

I'd love clarity on 2) as it seems like that could be a while down the road.

Reply:
Number 2 is correct. Since we don't ship the Razor EA, we have to wait for any changes to it to be inserted into VS main. That normally doesn't take _too_ long though. This is different to Roslyn where if you sometimes have to wait for something to be in a public preview, we only need to wait for main because our integration tests run on main, and we assume all team members work on VS main/IntPreview.